Project updates/Remove Locale source integration

friederbluemle opened the pull request:
**Fix Gradle error for fresh clones**
This allows anyone to clone the repo and immediately run `./gradlew tasks` (or `./gradlew build`). Previously this would result in an error when Gradle tried to access the non-existing file key.properties.

**Fix Lint errors**
A few Lint errors were fixed: Use `app` namespace (instead of `android`) for `showAsAction`. Add `@TargetApi` annotations (API calls were already correctly conditionalized)

**Replace Locale source integration with Gradle dependency**
This removes the entire `libraries` folder (with the `locale-api` sub-folder/project) by replacing the Locale API layer source code integration with a Gradle dependency.

Let me know if you prefer to have more than one pull request for this.

Asfaloth commented:
Looks good, but the translations will be overriden by the buildbot. Those dialects seem to have some issues since the language id transifex provides is wrong (missing `r`). This should be fixed in the buildbot. Can you exclude this commit in this PR? If you like, you can provide a PR for the buildbot to fix the directory name.

The other commits are fine.

friederbluemle commented:
Ah, good point about the auto-imported translation folders. I thought about it when I renamed the folder, but wasn't sure where exactly they are created/updated. I removed a few commits from this PR and will resubmit them as a new PR, once the buildbot is fixed. Please check https://github.com/clementine-player/Buildbot/pull/1

Asfaloth commented on build.gradle:
```diff
@@ -17,6 +17,7 @@ ext {
 allprojects {
     repositories {
         jcenter()
+        maven { url 'https://dl.bintray.com/twofortyfouram/maven' }
```
Do we really need this? `jcenter()` provides the bintray maven repo iirc.

friederbluemle replied:
Yes, this is needed, the twofortyfouram library is not available from `jcenter()`

Asfaloth commented:
It's good to know that the locale api is now in a maven repo :-) If the additional maven repo is really needed, this can be merged.

friederbluemle commented:
Unfortunately it's needed, otherwise the dependency cannot be resolved.

friederbluemle commented:
See also the [Gradle section of android-plugin-api-for-locale's README.md](https://github.com/twofortyfouram/android-plugin-api-for-locale#gradle)
We might be able to remove this additional maven repo in the future, if android-plugin-api-for-locale gets deployed to regular jcenter.

Asfaloth commented:
Okay, thanks for the feedback.
